### ISSUE TITLE
generated .../$types.d.ts has imports with '.ts'-extension which causes the new +page.server.ts/+page.svelte data prop to have type 'any'

### ISSUE THREAD

**Issue by ollema** [sveltejs/kit#5899]
update: see https://github.com/sveltejs/kit/issues/5899#issuecomment-1215987727 for the actual issue

---

### Describe the bug

First off - maybe this is expected? But when I read:
https://kit.svelte.dev/docs/routing#$types
> Throughout the examples above, we've been importing types from a $types.d.ts file. This is a file SvelteKit creates for you in a hidden directory if you're using TypeScript (or JavaScript with JSDoc type annotations) to give you type safety when working with your root files.
> 
> For example, annotating export let data with PageData (or LayoutData, for a +layout.svelte file) tells TypeScript that the type of data is whatever was returned from load:

I thought that the `data` prop would be fully typed!

With the `+page.server.ts`:
```ts
import type { PageServerLoad } from './$types';

import { getEvent } from '$lib/api';

export const load: PageServerLoad = async ({ params }) => {
	const { event, eventResponses } = await getEvent(params.id);

	return {
		event: event,
		eventResponses: eventResponses
	};
};
```
and `+page.svelte`:
```svelte
<script lang="ts">
	import type { PageData } from './$types';

	export let data: PageData;
</script>
```
I expected `data` to have the type
```
data: {
    event: event,
    eventResponses: eventResponses
}
```
but instead, as noted earlier, I get `data: any` (hope you understand the notation used)

### Reproduction

see the description

can also clone or checkout my repo:
https://github.com/ollema/sersophane/tree/b1380f17998c430e7f91404f92d6fed21e37c001/src/routes/events/%5Bid%5D

I think this is not super hard to reproduce - either it should happen for everyone OR I might have just misinterpreted the $types chapter

### Logs
I get this "problem" from vscode when viewing `/home/s0001325/work/sersophane/tsconfig.json`:

```
[{
    "resource": "/home/s0001325/work/sersophane/tsconfig.json",
    "owner": "typescript",
    "severity": 8,
    "message": "Cannot write file '/home/s0001325/work/sersophane/src/routes/+page.js' because it would overwrite input file.",
    "source": "ts",
    "startLineNumber": 1,
    "startColumn": 1,
    "endLineNumber": 1,
    "endColumn": 2
}]
```
Tried restarting vscode, tried reinstalling packages etc. it is probably not related but it was not there before the migration. Who knows!

My `tsconfig.json` is very simple:
```
{
	"extends": "./.svelte-kit/tsconfig.json",
	"compilerOptions": {
		"allowJs": true,
		"checkJs": true,
		"esModuleInterop": true,
		"forceConsistentCasingInFileNames": true,
		"resolveJsonModule": true,
		"skipLibCheck": true,
		"sourceMap": true,
		"strict": true
	}
}
```

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 20.04.4 LTS (Focal Fossa)
    CPU: (16) x64 Intel(R) Core(TM) i9-10885H CPU @ 2.40GHz
    Memory: 19.86 GB / 31.20 GB
    Container: Yes
    Shell: 5.0.17 - /bin/bash
  Binaries:
    Node: 18.7.0 - ~/.nvm/versions/node/v18.7.0/bin/node
    npm: 8.15.0 - ~/.nvm/versions/node/v18.7.0/bin/npm
  Browsers:
    Chrome: 104.0.5112.79
  npmPackages:
    @sveltejs/adapter-node: next => 1.0.0-next.86 
    @sveltejs/kit: next => 1.0.0-next.407 
    svelte: ^3.49.0 => 3.49.0 
    vite: ^3.0.7 => 3.0.7
```

### Severity

annoyance

**Comment by ollema** [comment]
(also, this post: https://github.com/sveltejs/kit/discussions/5774#discussioncomment-3292699 in the migration guide refers to `ServerLoad` when it probably should refer to `PageServerLoad`)

**Comment by ollema** [comment]
it seems related to the file extensions used in the imports of the generated types in `.svelte-kit`.

currently, I have
```ts
import type * as Kit from '@sveltejs/kit';

interface RouteParams extends Partial<Record<string, string>> { id: string }

export type Errors = null;
export type PageData = Kit.AwaitedProperties<Awaited<ReturnType<typeof import('./proxy+page.server.ts').load>>>;
export type PageServerData = Kit.AwaitedProperties<Awaited<ReturnType<typeof import('./proxy+page.server.ts').load>>>;
export type PageServerLoad = Kit.ServerLoad<RouteParams, import('../../$types.js').LayoutServerData>;
export type Action = Kit.Action<RouteParams>
```
if I remove the `.ts` extension it works:
```ts
import type * as Kit from '@sveltejs/kit';

interface RouteParams extends Partial<Record<string, string>> { id: string }

export type Errors = null;
export type PageData = Kit.AwaitedProperties<Awaited<ReturnType<typeof import('./proxy+page.server').load>>>;
export type PageServerData = Kit.AwaitedProperties<Awaited<ReturnType<typeof import('./proxy+page.server').load>>>;
export type PageServerLoad = Kit.ServerLoad<RouteParams, import('../../$types.js').LayoutServerData>;
export type Action = Kit.Action<RouteParams>
```
and now `data` is typed!

**Comment by ollema** [comment]
@dummdidumm not sure if this is working as intended, I still get imports with the `.ts` extension.

tried to remove `.svelte-kit`, `node_modules` and `pnpm_lock.yaml` and did a clean install with `pnpm install` followed by `pnpm run check` and `pnpm run build`. not sure what triggers the creation of these files

**Comment by iceghost** [comment]
@ollema yes, the fix at #5907 missed the `AwaitedProperties` 😅 i think someone will get to it soon!
